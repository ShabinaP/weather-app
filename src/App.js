import React, { useState, useEffect } from "react";
import getForecast from "./requests/getForecast";
import LocationDetails from "./Components/LocationDetails";
import SearchForm from "./Components/SearchForm";
import ForecastSummaries from "./Components/ForecastSummaries";
import ForecastDetails from "./Components/ForecastDetails";
import "./styles/App.css";
import "./styles/index.css";

const App = () => {
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [searchText, setSearchText] = useState("");
  const [selectedDate, setSelectedDate] = useState(0);
  const [errorMessage, setErrorMessage] = useState("");
  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );
  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };
  const handleCitySearch = () => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  };

  useEffect(() => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  }, []);
  return (
    <div className="weather-app">
      <LocationDetails
        city={location.city}
        country={location.country}
        errorMessage={errorMessage}
      />
      <SearchForm
        onSubmit={handleCitySearch}
        searchText={searchText}
        setSearchText={setSearchText}
      />
      {!errorMessage && (
        <>
          <ForecastSummaries
            forecasts={forecasts}
            onForecastSelect={handleForecastSelect}
          />
          {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
        </>
      )}
    </div>
  );
};

export default App;
