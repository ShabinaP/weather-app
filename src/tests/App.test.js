import { render, screen } from "@testing-library/react";
import React from "react";
import App from "../components/App";
import { location, forecasts } from "../data/forecast.json";

test("renders correctly", () => {
  const { asFragment } = render(
    <App forecasts={forecasts} location={location} />
  );

  expect(asFragment()).toMatchSnapshot();
});
