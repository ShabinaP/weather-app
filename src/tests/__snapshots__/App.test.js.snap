// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1
      class="location-details"
    >
      London, UK
    </h1>
    <div
      class="forecast-summaries"
    >
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          1525046400000
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <i
            class="wi wi-day-sunny"
            name="owm"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          11°C
        </div>
        <div
          class="forecast-summary__description"
        >
          Clear
        </div>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          1525132800000
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <i
            class="wi wi-day-thunderstorm"
            name="owm"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          13°C
        </div>
        <div
          class="forecast-summary__description"
        >
          Stormy
        </div>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          1525219200000
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <i
            class="wi wi-day-snow"
            name="owm"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          1°C
        </div>
        <div
          class="forecast-summary__description"
        >
          Heavy Snow
        </div>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          1525305600000
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <i
            class="wi wi-tornado"
            name="owm"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          20°C
        </div>
        <div
          class="forecast-summary__description"
        >
          Tornado
        </div>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary__date"
        >
          1525392000000
        </div>
        <div
          class="forecast-summary__icon"
          data-testid="forecast-icon"
        >
          <i
            class="wi wi-day-haze"
            name="owm"
          />
        </div>
        <div
          class="forecast-summary__temperature"
        >
          25°C
        </div>
        <div
          class="forecast-summary__description"
        >
          Hazy
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
